pkg/vsphere/datastore/datastore.go

```diff
@@ -89,12 +88,13 @@ func NewHelperFromURL(ctx context.Context, s *session.Session, u *url.URL) (*Hel
 	}
 
 	d := &Helper{
-		ds:      vsDs,
-		s:       s,
-		fm:      fm,
-		RootURL: u.Path,
+		ds: vsDs,
+		s:  s,
+		fm: fm,
 	}
 
+	d.RootURL.FromString(u.Path)
+
 	return d, nil
 }
 
```

Comment:
I'm renegging on this.  Changing the constructor from URL to DatastorePath is the right call, but the trickle is too large for this specific bug fix.  I could just make a new constructor from datastore path and leave the from url constructor, but that doesn't do very much if nothing is calling it.  So I'm going to punt on adding this for now and get back to it after this PR.  I just want to fix the issue mentioned in the fix.  The rest is just fixing debt and i don't want to conflate the two and blow out my point calculation for the story.